conans/test/integration/toolchains/gnu/test_pkgconfigdeps.py

```diff
@@ -725,3 +725,48 @@ def generate(self):
     client.save({"conanfile.py": conanfile}, clean_first=True)
     client.run("install . -pr:h default -pr:b default", assert_error=True)
     assert "The packages ['tool'] exist both as 'require' and as 'build require'" in client.out
+
+
+def test_error_missing_pc_build_context():
+    """
+    PkgConfigDeps was failing, not generating the zlib.pc in the
+    build context, for a test_package that both requires(example/1.0) and
+    tool_requires(example/1.0), which depends on zlib
+    # https://github.com/conan-io/conan/issues/12664
+    """
+    c = TestClient()
+    example = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        class Example(ConanFile):
+            name = "example"
+            version = "1.0"
+            requires = "game/1.0"
+            generators = "PkgConfigDeps"
+            settings = "build_type"
+            def build(self):
+                assert os.path.exists("math.pc")
+                assert os.path.exists("engine.pc")
+                assert os.path.exists("game.pc")
+            """)
+    c.save({"math/conanfile.py": GenConanfile("math", "1.0").with_settings("build_type"),
+            "engine/conanfile.py": GenConanfile("engine", "1.0").with_settings("build_type")
+                                                                .with_require("math/1.0"),
+            "game/conanfile.py": GenConanfile("game", "1.0").with_settings("build_type")
+                                                            .with_requires("engine/1.0"),
+            "example/conanfile.py": example,
+            "example/test_package/conanfile.py": GenConanfile().with_requires("example/1.0")
+                                                               .with_build_requires("example/1.0")
```

Comment:
@ericLemanissier here it is: `.with_build_requires("example/1.0")` So I think it's well-tested, doesn't it?

Reply:
indeed, I missed it because I searched for `tool_requires`